app/src/main/java/org/openobservatory/ooniprobe/common/Application.java

```diff
@@ -35,24 +38,39 @@ public class Application extends android.app.Application {
 	@Inject Gson _gson;
 	@Inject OkHttpClient _okHttpClient;
 	@Inject OONIAPIClient _apiClient;
+	ExecutorService executorService = Executors.newFixedThreadPool(4);
 
 	public AppComponent component;
 
 	@Override public void onCreate() {
 		super.onCreate();
+		TimingLogger timings = new TimingLogger(Application.class.getSimpleName(), "app-on-create");
 		component = buildDagger();
 		component.inject(this);
+		timings.addSplit("dagger-init");
 
 		FlowManager.init(this);
 		if (BuildConfig.DEBUG)
 			FlowLog.setMinimumLoggingLevel(FlowLog.Level.V);
+		timings.addSplit("flow-init");
 		AppLifecycleObserver appLifecycleObserver = new AppLifecycleObserver();
 		ProcessLifecycleOwner.get().getLifecycle().addObserver(appLifecycleObserver);
 		Measurement.deleteOldLogs(this);
 		ThirdPartyServices.reloadConsents(this);
 
+		timings.addSplit("lifecycle-init");
+		executorService.execute(() -> {
+			if (_preferenceManager.canCallDeleteJson())
+				Measurement.deleteUploadedJsons(Application.this);
+			Measurement.deleteOldLogs(Application.this);
+		});
```

Comment:
I am not completely convinced by running this step in parallel and I may need additional data to understand whether this change is actually thread safe. Is the code that we're calling inside this lambda thread safe or synchronized. If not, we may be causing a bunch of data races in this PR.

Reply:
Hi @bassosimone  this code segment is meant to be run in another thread other than the main thread.
The only thing it does is remove the timing from the main thread. Since no other process is accessing the `logs` and `json` files in the few seconds it uses to delete them, I don't see a problem with syncronization.

Reply:
Ok, sounds good!